### ISSUE TITLE
Cannot add a polygon field when this field is indexed with a lucene spatial engine

### ISSUE THREAD

**Issue by BrunoSupiot** [orientechnologies/orientdb#8610]
## Expected behavior  
Vertex properly created

## Actual behavior  
com.orientechnologies.orient.core.index.OIndexException: Cannot create the index 'Agent.polygonLocation' DB name="test" --> java.lang.ClassCastException: java.lang.Float cannot be cast to java.util.List

## Steps to reproduce  
CREATE CLASS Agent EXTENDS V
CREATE PROPERTY Agent.polygonLocation EMBEDDED OPolygon;
CREATE INDEX Agent.polygonLocation ON Agent(polygonLocation) SPATIAL ENGINE LUCENE;
INSERT INTO Agent SET polygonLocation = {"@class":"OPolygon","coordinates":[[1.399657130241394,43.58523264988069],[1.400507390499115,43.58523264988069],[1.400507390499115,43.58608361522166],[1.399657130241394,43.58608361522166],[1.399657130241394,43.58523264988069]]} 

Note that if the index is not created, then the vertex is created.
If the index is created after, the error appears when the index is created

**Comment by wolf4ood** [comment]
Hi @BrunoSupiot 

the syntax for creating polygons it's the same as GeoJSON

https://geojson.org/geojson-spec.html#polygon


So in your case with a Single Ring should be 

```
INSERT INTO Agent SET polygonLocation = {"@Class":"OPolygon","coordinates":[[[1.399657130241394,43.58523264988069],[1.400507390499115,43.58523264988069],[1.400507390499115,43.58608361522166],[1.399657130241394,43.58608361522166],[1.399657130241394,43.58523264988069]]]}
```

**Comment by wolf4ood** [comment]
Hi @BrunoSupiot 

that is strange , work on my box with 3.0.9 and this script sequence

```
CREATE CLASS Agent EXTENDS V
CREATE PROPERTY Agent.polygonLocation EMBEDDED OPolygon;
CREATE INDEX Agent.polygonLocation ON Agent(polygonLocation) SPATIAL ENGINE LUCENE;
INSERT INTO Agent SET polygonLocation = {"@Class":"OPolygon","coordinates":[[[1.399657130241394,43.58523264988069],[1.400507390499115,43.58523264988069],[1.400507390499115,43.58608361522166],[1.399657130241394,43.58608361522166],[1.399657130241394,43.58523264988069]]]}
```

**Comment by BrunoSupiot** [comment]
I just tried with 3.09 version of OrientDB, and it works!
So the bug was corrected (directly or indirectly) with the 3.09.
Thank you for your help

Bruno
